spacepy/datamodel.py

```diff
@@ -1337,12 +1337,23 @@ def readJSONMetadata(fname, **kwargs):
     if hasattr(fname, 'read'):
         lines = fname.read()
     else:
-        with open(fname, 'r') as f:
-            lines = f.read()
+        # also possible in an exploration sense utilizing UnicodeDecodeError
+        if fname.endswith('.gz'):
+            import gzip
+            gzh = gzip.GzipFile(filename=fname)
+            lines = gzh.read()
+            gzh.close()
```

Comment:
ok 2.6, I will add the context manager.

Reply:
These are being shown in a really weird order (probably order I poked at them rather than file order) and I might have messed up...adding these kwargs here opens the file in text mode so you don't need the decode below.

Reply:
Updated, this is a nice solution

Reply:
@jtniehof , I'm confused here. 

`encoding` does not go through to `gzip.GzipFile`

`TypeError: __init__() got an unexpected keyword argument 'encoding'`

Reply:
changed `gzip.GzipFile` to `gzip.open` that works